common/web/gesture-recognizer/src/engine/headless/gestures/matchers/pathMatcher.ts

```diff
@@ -0,0 +1,85 @@
+import { SimpleGestureSource } from "../../simpleGestureSource.js";
+import { ContactModel, PointModelResolution } from "../specs/contactModel.js";
+import { ManagedPromise, TimeoutPromise } from "@keymanapp/web-utils";
+
+export class PathMatcher<Type> {
+  private timerPromise?: TimeoutPromise;
+  private model: ContactModel;
+  private source: SimpleGestureSource<Type>;
+
+  private readonly publishedPromise: ManagedPromise<PointModelResolution>
+
+  public get promise() {
+    return this.publishedPromise.corePromise;
+  }
+
+  constructor(model: ContactModel, source: SimpleGestureSource<Type>) {
+    this.model = model;
+    this.publishedPromise = new ManagedPromise<PointModelResolution>();
+    this.source = source;
+
+    /* c8 ignore next 3 */
+    if(!model || !source) {
+      throw new Error("A gesture-path source and contact-path model must be specified.");
+    }
```

Comment:
Shouldn't this be a precondition before assigning class members? Not that it really matters but preconditions normally come first in a function.